examples/js/CSVnVRTParser.js

```diff
@@ -0,0 +1,207 @@
+
+/**
+ * A module to parse OGR Virtual Format files.
+ *
+ * See the [GDAL documentation](https://gdal.org/drivers/vector/vrt.html) and
+ * the [xsd
+ * schema](https://github.com/OSGeo/gdal/blob/master/gdal/data/ogrvrt.xsd) of
+ * the OGR VRT file.
+ *
+ * @example
+ * Fetcher.multi('data', {
+ *     xml: ['vrt'],
+ *     text: ['csv']
+ * }).then(function _(res) {
+ *     return CSVnVRTParser.parse(res, {
+ *         buildExtent: true,
+ *         crsOut: 'EPSG:4326'
+ *     });
+ * }).then(function _(feature) {
+ *     var source = new itowns.FileSource({ parsedData: feature });
+ *     var layer = new itowns.ColorLayer('CSVnVRT', { source });
+ *     view.addLayer(layer);
+ * });
+ *
+ * @module CSVnVRTParser
+ */
+var CSVnVRTParser = (function _() {
+    var coord = new itowns.Coordinates('EPSG:4326');
+
+    function csv2json(csv) {
+        var lines = csv.trim().split('\n');
+        for (var i = 0; i < lines.length; i++) {
+            lines[i] = lines[i].split(/;|\t|,/);
+        }
+
+        return {
+            header: lines.shift(),
+            content: lines
+        };
+    }
+
+    function xml2json(xml, json) {
+        var res = {};
+
+        var attributes = xml.getAttributeNames();
+        if (attributes.length > 0) {
+            res.attributes = {};
+            for (var i = 0; i < attributes.length; i++) {
+                res.attributes[attributes[i]] = xml.getAttributeNode(attributes[i]).value;
+            }
+        }
+
+        if (xml.childElementCount > 0) {
+            for (var i = 0; i < xml.childElementCount; i++) {
+                xml2json(xml.children[i], res);
+            }
+        } else if (xml.textContent) {
+            res.value = xml.textContent;
+        }
+
+        var name = xml.nodeName;
+
+        if (!json[name]) {
+            json[name] = res;
+        } else if (json[name].length > 0) {
+            json[name].push(res);
+        } else {
+            json[name] = [res];
+        }
+
+        return json;
+    }
+
+    function getGeometryType(type) {
+        switch (type) {
+            case 'wkbPoint':
+            case 'wkbMultiPoint':
+                return itowns.FEATURE_TYPES.POINT;
+            case 'wkbLineString':
+            case 'wkbMultiLineString':
+                return itowns.FEATURE_TYPES.LINE;
+            case 'wkbPolygon':
+            case 'wkbMultiPolygon':
+                return itowns.FEATURE_TYPES.POLYGON;
+            default:
+                throw new Error('This type of GeometryType is not supported yet: ' + type);
+                break;
+        }
+    }
+
+    function OGRVRTLayer2Feature(layer, data, crs, options) {
+        var crs = (layer.LayerSRS && layer.LayerSRS.value) || crs;
+
+        var type = itowns.FEATURE_TYPES.POINT;
+        if (layer.GeometryType) {
+            type = getGeometryType(layer.GeometryType.value);
+        }
+
+        var feature = new itowns.Feature(type, options.crsOut, options);
+
+        if (layer.Field) {
```

Comment:
Could you comment this part?

Reply:
Do you want to have more precision on the properties read here ? I think the linked documentation (https://gdal.org/drivers/vector/vrt.html) should be ok, no ?

Reply:
yes